### ISSUE TITLE
Error generating rxn_maps due to mismatch in array size

### ISSUE THREAD

**Issue by starkAhmed43** [rxn4chemistry/rxnmapper#56]
Hello,

I am trying to generate atom mappings for some 30,000 reaction SMILES. These reaction SMILES are generated for the reactions from the Rhea database. I download the mol files of the participating metabolites for ChEBI, generate SMILES using RDKIT and then concatenate them using `.` and `>>` to generate reaction SMILES.

Using, `BatchedMapper` I am able to successfully generate atom maps for 24,600 of them. For the remaining 5000ish, when I use `RxnMapper` to identify the cause of error I get a few different error types:

```
Error: index X is out of bounds for axis 0 with size X
Error: could not broadcast input array from shape (A,) into shape (B,)
Error: The size of tensor a (D) must match the size of tensor b (512) at non-singleton dimension 1
```

Error 2 accounts for the vast majority of unsuccessful maps.

I am from a data science background and have zero chemistry knowledge and thus, while I understand the error messages, I do not understand what is causing them.

It would be really great if the authors of `rxnMapper` could help me out here. Please let me know if you need more details regarding the errors I am getting.

**Comment by avaucher** [comment]
Hi, 
I agree that the error messages are sometimes cryptic. Is it possible for you to share a few of these reactions?

My best guess is that the reaction SMILES strings are too long for the current model, but happy to check directly.

**Comment by starkAhmed43** [comment]
SMILES examples for:

Error 1:

Error 2: could not broadcast input array from shape (A,) into shape (B,)

*N[C@@H](CCC(N)=O)C(=O)[O-].O>>*N[C@@H](CCC(=O)[O-])C(=O)[O-].[NH4+]

*[C@H]1C[C@H](OP(=O)([O-])[O-])[C@@H](CO)O1.O>>*[C@H]1C[C@H](O)[C@@H](CO)O1.O=P([O-])([O-])O

*N[C@@H](COP(=O)([O-])OCC(C)(C)[C@@H](O)C(=O)NCCC(=O)NCCSC(*)=O)C(*)=O.C[S+](CC[C@H]([NH3+])C(=O)[O-])C[C@H]1O[C@@H](n2cnc3c(N)ncnc32)[C@H](O)[C@@H]1O>>*C(=O)N[C@H]1CCOC1=O.[H+].*N[C@@H](COP(=O)([O-])OCC(C)(C)[C@@H](O)C(=O)NCCC(=O)NCCS)C(*)=O.CSC[C@H]1O[C@@H](n2cnc3c(N)ncnc32)[C@H](O)[C@@H]1O

*C(=O)OC1CCC2(C)C(CCC3C4CCC(*)C4(C)CCC32)C1.O>>*C(=O)[O-].*C1CCC2C3CCC4CC(O)CCC4(C)C3CCC12C.[H+]

**Comment by avaucher** [comment]
Thanks for the examples!

I started looking into it and will shortly have a fix for Errors 1 and 2. This was caused by the presence of asterisks in the reaction SMILES, which sometimes stand for atom placeholders. After the fix, these reactions should succeed.

Error 3 seems to be caused by too long reaction SMILES. I'll improve the error message for these.

**Comment by avaucher** [comment]
Errors 1 and 2 fixed by #57.

**Comment by avaucher** [comment]
The new version on PyPi (0.4.0) should cover all the above! Feel free to reopen this issue if needed.
